src/main/java/subway/repository/JdbcLineRepository.java

```diff
@@ -0,0 +1,99 @@
+package subway.repository;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import subway.dao.LineDao;
+import subway.dao.SectionDao;
+import subway.domain.Line;
+import subway.domain.Section;
+import subway.domain.Station;
+import subway.entity.LineEntity;
+import subway.entity.SectionEntity;
+import subway.exception.NoSuchLineException;
+import subway.mapper.LineMapper;
+import subway.mapper.SectionMapper;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+@Transactional
+public class JdbcLineRepository implements LineRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final LineDao lineDao;
+    private final SectionDao sectionDao;
+
+    private final RowMapper<Section> sectionRowMapper = (rs, rowNum) ->
+            new Section(
+                    rs.getLong(1),
+                    new Station(rs.getLong(2), rs.getString(3)),
+                    new Station(rs.getLong(4), rs.getString(5)),
+                    rs.getInt(6)
+            );
+
+    public JdbcLineRepository(final JdbcTemplate jdbcTemplate, final LineDao lineDao, final SectionDao sectionDao) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.lineDao = lineDao;
+        this.sectionDao = sectionDao;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public Line findById(final long lineId) {
+        LineEntity lineEntity = lineDao.findById(lineId)
+                .orElseThrow(() -> new NoSuchLineException(lineId));
+
+        String sql = "SELECT section.id, up.id, up.name, down.id, down.name, section.distance FROM section "
+                + "INNER JOIN station up ON section.up_station_id = up.id "
+                + "INNER JOIN station down ON section.down_station_id = down.id "
+                + "WHERE section.line_id = ?";
+
+        List<Section> sections = jdbcTemplate.query(sql, sectionRowMapper, lineId);
+
+        return new Line(lineEntity.getId(), lineEntity.getName(), lineEntity.getColor(), sections);
+    }
+
+    @Override
+    public void updateSections(final Line line) {
+        sectionDao.deleteAllByLineId(line.getId());
+
+        List<SectionEntity> sections = line.getSections()
+                .stream()
+                .map(section -> SectionMapper.toEntity(line.getId(), section))
+                .collect(Collectors.toList());
+
+        sectionDao.insertAll(sections);
+    }
```

Comment:
삭제 후 새로 생성하는 방식이 아닌 update로 한다면 아래와 같은 방식으로 하면되지 않을까 싶네요 :)

```suggestion
  @Override
    public int updateSections(final List<Section> sections) {
        final String sql = "update SECTION set up_station_id = ?, down_station_id = ?, distance = ? where id = ?";
        return jdbcTemplate.batchUpdate(sql,
                new BatchPreparedStatementSetter() {
                    @Override
                    public void setValues(final PreparedStatement ps, final int i) throws SQLException {
                        Section section = sections.get(i);
                        ps.setLong(1, section.getUpStation().getId());
                        ps.setLong(2, section.getDownStation().getId());
                        ps.setInt(3, section.getDistance());
                        ps.setLong(4, section.getId());
                    }

                    @Override
                    public int getBatchSize() {
                        return sections.size();
                    }
                }).length;
    }
```

주노가 하신대로 삭제 후 재생성하는 방식이 더 간단할 것 같기도 하네요! 서울 지하철도 노선은 11개, 역도 289개 밖에 되지 않아서 성능 문제도 크게 걱정할게 없어 보입니다. 잘 구현하셨네요 👍